### PR TITLE
ticket 4 produces results

### PR DESCRIPTION
**Pull request by bigsegs**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -35,4 +35,22 @@ describe("GET /api/topics ", () => {
...
+
+  describe('GET /api/articles/:article_id',()=>{
+    it('should return 200 and the correct article',()=>{
+      return request(app)
+      .get('/api/articles/1')
+      .expect(200)
+      .then((result)=>{
+        console.log("result in test",result.body);
+        expect(result.body.article).toHaveProperty('title');
+        expect(result.body.article).toHaveProperty('article_id');
```
Comment (xViolaine) [inline __tests__/app.test.js:47]: Ideally, we shouldn't just be testing whether the property exists, but if it's the article ID we expect to get back

File: app.js [hunk 2]
```diff
@@ -1,12 +1,15 @@
 const express=require('express');
 const app = express();
 const {getTopics}=require('./topicController.js');
-
-
+const {getArticleById}=require('./articleController.js');
+app.use(express.json());
```
Comment (xViolaine) [inline app.js:5]: Do we really need app.use(express.json()) here already? We should only add code if it makes sense to be there.

File: __tests__/app.test.js [hunk 3]
```diff
@@ -35,4 +35,22 @@ describe("GET /api/topics ", () => {
...
-  
\ No newline at end of file
+
+  describe('GET /api/articles/:article_id',()=>{
+    it('should return 200 and the correct article',()=>{
+      return request(app)
+      .get('/api/articles/1')
+      .expect(200)
+      .then((result)=>{
+        console.log("result in test",result.body);
```
Comment (xViolaine) [inline __tests__/app.test.js:45]: Really good to see you use console.logs to see whether everything is pulling through the way we expect it to, but we should remove unneeded console.logs before we push our code up and create a PR - I'm only highlighting this instance, but there are quite a few in your code!

File: __tests__/app.test.js [hunk 4]
```diff
@@ -35,4 +35,22 @@ describe("GET /api/topics ", () => {
...
+        expect(result.body.article).toHaveProperty('article_id');
+        expect(result.body.article).toHaveProperty('body');
+        expect(result.body.article).toHaveProperty('topic');
+        expect(result.body.article).toHaveProperty('created_at');
+        expect(result.body.article).toHaveProperty('votes');
+        expect(result.body.article).toHaveProperty('article_img_url');
+      })
+    })
+  })
+
```
Comment (xViolaine) [inline __tests__/app.test.js:56]: We need to write tests and code to make sure any potential errors are handled. Hint - there are two cases we need to consider here, 1) an invalid ID (for example, a string, not a number) and 2) and ID that is valid, but non-existent (for example, ID 9999)

**Review by xViolaine — CHANGES_REQUESTED** [review]

**Review by xViolaine — CHANGES_REQUESTED** [review]
Nicely done so far, but please make sure to add error handling (as well as testing for it) and remove any unnecessary lines of code! Make sure to resubmit your PR once done so.

File: __tests__/app.test.js (hunk 3 above)
Comment (bigsegs) [reply __tests__/app.test.js:45]: ah sorry was last night trying to catch up forgot to remove

File: __tests__/app.test.js (hunk 1 above)
Comment (bigsegs) [reply __tests__/app.test.js:47]: done for all

Comment (bigsegs) [reply __tests__/app.test.js:47]: I'm not sure if the timestamp I get back from the query is what is required here? If I sql the query separately I get something slightly different and I'm 'assuming' that this is the JSON version? So i just copied from the result and used it as 'expected' believing it to be correct in this instance?

File: __tests__/app.test.js (hunk 4 above)
Comment (bigsegs) [reply __tests__/app.test.js:56]: think i got that right, had to parse the id as it came in as a string anyway even though it was a number?

File: __tests__/app.test.js [hunk 5]
```diff
@@ -35,4 +35,42 @@ describe("GET /api/topics ", () => {
...
\ No newline at end of file
+
+  describe('GET /api/articles/:article_id',()=>{
+    it('should return 200 and the correct article',()=>{
+      return request(app)
+      .get('/api/articles/1')
+      .expect(200)
+      .then((result)=>{
+        expect(result.body.article.title).toEqual('Living in the shadow of a great man');
+        expect(result.body.article.article_id).toEqual(1);
```
Comment (pippaamy) [inline __tests__/app.test.js:46]: great job here! but the only one we need to check is the id as the others could change but the Id will stay the same!

**Review by pippaamy — COMMENTED** [review]
great job!

**Review by pippaamy — COMMENTED** [review]
great job!

**Pull request merged**